### ISSUE TITLE
Unary minus operator is parsed with too high precedence

### ISSUE THREAD

**Issue by leftaroundabout** [bmillwood/haskell-src-meta#64]
```haskell
Prelude Language.Haskell.Meta.Parse> let Right nq = parseExp "-3^2"
Prelude Language.Haskell.Meta.Parse> nq
UInfixE (AppE (VarE GHC.Num.negate) (LitE (IntegerL 3))) (VarE ^) (LitE (IntegerL 2))
Prelude Language.Haskell.Meta.Parse> $(pure nq)
9
```
...but the correct value would obviously be
```haskell
Prelude Language.Haskell.Meta.Parse> -3^2
-9
```
Basically, `haskell-src-meta` seems to treat the unary minus almost<sup>†</sup> like an ordinary function application – with higher precedence than any infix. Actually, Haskell's unary minus has only the fixity of an ordinary addition/subtraction, i.e. `infixl 6`.

Worth noting that this does not happen when directly using `haskell-src-exts`:
```haskell
Prelude Language.Haskell.Exts.Parser> parseExp "-3^2"
ParseOk (NegApp (InfixApp (Lit (Int 3)) (QVarOp (UnQual (Symbol "^"))) (Lit (Int 2))))
```

<hr>

<sup>†</sup><sub>Not _quite_ like function application: something like `- succ 1` is correctly parsed as `- (succ 1)`.</sub>

**Comment by JonasDuregard** [comment]
This is apparently because the default ParseMode does not include the fixity of `^` (which defaultParseMode from HSE does). This makes #65 more urgent.

**Comment by JonasDuregard** [comment]
This fixes the bug for your example: 8bfa70a501ed048298ecf992f613b3237631c98a
